xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/doc/XWikiDocument.java

```diff
@@ -1941,16 +1941,33 @@ public boolean isContentDirty()
         return this.isContentDirty;
     }
 
+    /**
+     * Increment the current document version.
+     * This method will use {@link #getNextVersion(Version, boolean)} to compute the new version.
+     */
     public void incrementVersion()
     {
-        if (this.version == null) {
-            this.version = new Version("1.1");
+        this.version = getNextVersion(this.version, isMinorEdit());
+    }
+
+    /**
+     * This method computes the next version and returns it, but won't change the current version.
```

Comment:
Not sure all this warning is required for a `static` method starting with `get` and taking as parameter the version to increment but OK :)